aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1893,6 +1951,145 @@ std::tuple<Tensor, Tensor> _triangular_solve_helper_cuda(const Tensor& self, con
+  auto T_sizes = tau.sizes().vec();
+  T_sizes.push_back(k);
+  Tensor T = at::zeros(T_sizes, self.options());
+  T.transpose_(-2, -1);  // make T column-major
+  auto ldt = std::max<magma_int_t>(1, k);
+  auto T_matrix_stride = matrixStride(T);
+  auto T_data = T.data_ptr<scalar_t>();
+
+  scalar_t** self_array;
+  scalar_t** T_array;
+  scalar_t** tau_array;
+
+  ALLOCATE_ARRAY(self_array, scalar_t*, batchsize);
+  ALLOCATE_ARRAY(T_array, scalar_t*, batchsize);
+  ALLOCATE_ARRAY(tau_array, scalar_t*, batchsize);
+
+  // Set up the created arrays
+  for (int64_t i = 0; i < batchsize; i++) {
+    self_array[i] = &self_data[i * self_matrix_stride];
+    T_array[i] = &T_data[i * T_matrix_stride];
+    tau_array[i] = &tau_data[i * tau_stride];
+  }
+
+  // make sure that all elements of elementary reflectors including 0's and 1's are stored, unlike LAPACK
+  self.diagonal(/*offset=*/0, /*dim1=*/-2, /*dim2=*/-1).fill_(1);
+  self.tril_();
+
+  // Now compute T matrix using magmaLarftBatched
+  if (batchsize > 0) {
+    MAGMAQueue magma_queue(self.get_device());
+
+    constexpr int64_t batch_limit = 65535;
+    // Compute as many batches of 65535 possible
+    // The number of "mini"-batches are floor(batchsize / batch_limit)
+    // and these cover floor(batchsize / batch_limit) * batch_limit
+    int64_t mini_batches = batchsize / batch_limit, mini_idx;
+    for (mini_idx = 0; mini_idx < mini_batches * batch_limit; mini_idx += batch_limit) {
+      scalar_t** self_array_cur = &self_array[mini_idx];
+      scalar_t** tau_array_cur = &tau_array[mini_idx];
+      scalar_t** T_array_cur = &T_array[mini_idx];
+      magmaLarftBatched<scalar_t>(m, k, 0, self_array_cur, lda, tau_array_cur, T_array_cur, ldt, &work_data, lwork, batch_limit, magma_queue);
+    }
+
+    // Compute whatever is left = batchsize - floor(batchsize / batch_limit) * batch_limit
+    // which concisely is equal to batchsize % batch_limit
+    if (batchsize % batch_limit != 0) {
+      magmaLarftBatched<scalar_t>(m, k, 0, &self_array[mini_idx], lda, &tau_array[mini_idx], &T_array[mini_idx], ldt, &work_data, lwork, batchsize % batch_limit, magma_queue);
+    }
+  }
+
+  // Compute result = (I - V T V^H) Ic,
+  // where I is an m x m identity matrix, Ic is an m x n identity matrix,
+  // V is an m x k matrix of reflectors and T is a k x k matrix
+  // With LAPACK same result could be obtained using LARFB
+  auto V = at::narrow(self, /*dim=*/-1, /*start=*/0, /*length=*/k);
+  auto W = at::narrow_copy(V, /*dim=*/-2, /*start=*/0, /*length=*/n);  // W <- Ic^H V: n x m @ m x k = n x k
+  at::matmul_out(W, T, W.conj().transpose(-2, -1));  // W <- T W^H: k x k @ k x n = k x n
+  at::matmul_out(self, V, W);  // self <- V W: m x k @ k x n = m x n
+  self.mul_(-1);
+  self.diagonal(/*offset=*/0, /*dim1=*/-2, /*dim2=*/-1).add_(1); // self <- self + Ic: m x n + m x n = m x n
+
+  // infos is not used here, make sure that it's filled with zeros
+  infos.fill_(0);
```

Comment:
Would you elaborate on why infos is unused in this case and why it needs to be filled with zeros here?

Reply:
I'll try to improve the comment. `infos` has to be filled with zeros because non-zero content would cause raising an error, alternatively, it could be initialized with zero values. MAGMA path here is more low-level implementation than LAPACK or cuSOLVER and there is no interface to pass `infos` argument to `magmaLarftBatched`.